backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt

```diff
@@ -973,3 +1018,50 @@ private fun ObjCExportNamer.ClassOrProtocolName.toNameAttributes(): List<String>
 
 private fun swiftNameAttribute(swiftName: String) = "swift_name(\"$swiftName\")"
 private fun objcRuntimeNameAttribute(name: String) = "objc_runtime_name(\"$name\")"
+
+interface TypeParameterProvider{
+    fun typeAvailable(kotlinType: KotlinType):Boolean
```

Comment:
I would prefer this method to be more like `fun getName(typeParameter: TypeParameterDescriptor): String?`
And this interface could be named like `ObjCExportScope` etc.

Reply:
Updated